picard/ui/options/renaming.py

```diff
@@ -173,22 +176,42 @@ def _example_to_filename(self, file):
                     if s_enabled and s_text:
                         parser = ScriptParser()
                         parser.eval(s_text, file.metadata)
-            filename = file.make_filename(file.filename, file.metadata, settings)
+            filename_before = file.filename
+            filename_after = file.make_filename(filename_before, file.metadata, settings)
             if not settings["move_files"]:
-                return os.path.basename(filename)
-            return filename
+                return os.path.basename(filename_before), os.path.basename(filename_after)
+            return filename_before, filename_after
         except ScriptError:
-            return ""
+            return "", ""
         except TypeError:
-            return ""
+            return "", ""
+
+    def _sample_example_files(self):
+        # Get a new sample of randomly selected files to use as renaming examples
+        import random
+        keys = list(self.tagger.files.keys())
+        files_len = len(keys)
+        indexes = set([random.randrange(0, files_len) for i in range(min(files_len, 10))])  # pick up to 10 samples
+        self._sampled_example_files = [self.tagger.files[keys[index]] for index in indexes]
+        self.update_examples()
```

Comment:
A much simpler (and reliable) way to do it:

```python
    def _sample_example_files(self):
        # Get a new sample of randomly selected files to use as renaming examples
        import random
        files = self.tagger.files
        length = min(len(files), 10)
        self._sampled_example_files = [files[key] for key in random.sample(files.keys(), k=length)]
        self.update_examples()
```

Reply:
The C programmer in me always like to access stuff with numerical indexes. xD